lua/nvim-treesitter/query_predicates.lua

```diff
@@ -85,5 +87,22 @@ query.add_predicate('has-type?', function(match, pattern, bufnr, pred)
   return vim.tbl_contains(types, node:type())
 end)
 
+
+-- Inject a language based on the value from `highlight.injections.{language}.{value}`.
```

Comment:
I don't think this should be under the highlight module since it will effect the injected language of the tree which affects more that just highlighting. For example, locals and refactor would be affected by this as well.

Reply:
so, is it ok to put this under a config-only module named `injections`?

Reply:
It would be great when you could derive the spot where this directive is configured by only looking at the query code. It could provide a path to the config variable or name a global/buffer variable.

Reply:
Like `inject-python-docstring!` (`inject-<lang>-docstring!`) and you could set the option via `b:python_docstring` or `g:python-docstring` (option and directive shared name)

Reply:
I like the idea of having per buffer options, but would we require users to set the global option too instead of setting it in the setup method? Or support both?